src/commands/eval.ts

```diff
@@ -196,6 +196,7 @@ export async function doEval(
         );
       }
     } else {
+      await eval_.delete();
```

Comment:
if the user requested not to save the results, we need to delete them

Reply:
Do we want to store this in memory entirely if the user does not want to save it? Thinking about the case where promptfoo crashes and this never gets called. May not be important.

Reply:
I like @mldangelo's idea of just using the in memory database when that option is enabled

Reply:
I just did it for real